nx/lib/nx/defn/expr.ex

```diff
@@ -163,7 +163,14 @@ defmodule Nx.Defn.Expr do
 
       zero =
         Composite.traverse(last, fn leaf ->
-          Nx.broadcast(0, Nx.devectorize(leaf).shape) |> Nx.vectorize(leaf.vectorized_axes)
+          vectorized_axes =
+            case leaf do
+              %{vectorized_axes: vectorized_axes} -> vectorized_axes
+              number when is_number(number) -> []
```

Comment:
I guess this could be anything, including a complex, so at best we should use is_tensor. But perhaps the best is to call `to_tensor(left)` before?

Reply:
You know, I thought about complex when writing this and forgot to test and add the case haha

Reply:
to_tensor(left) is the more uniform and easier solution